tests/gorepo/run.go

```diff
@@ -1315,3 +1310,65 @@ func getenv(key, def string) string {
 	}
 	return def
 }
+
+// splitQuoted splits the string s around each instance of one or more consecutive
+// white space characters while taking into account quotes and escaping, and
+// returns an array of substrings of s or an empty list if s contains only white space.
+// Single quotes and double quotes are recognized to prevent splitting within the
+// quoted region, and are removed from the resulting substrings. If a quote in s
+// isn't closed err will be set and r will have the unclosed argument as the
+// last element. The backslash is used for escaping.
+//
+// For example, the following string:
+//
+//	a b:"c d" 'e''f'  "g\""
+//
+// Would be parsed as:
+//
+//	[]string{"a", "b:c d", "ef", `g"`}
+//
+// [copied from src/go/build/build.go]
+func splitQuoted(s string) (r []string, err error) {
```

Comment:
this looks like a good candidate for some tests, but up to you

Reply:
It's a direct lift from the upstream: https://github.com/golang/go/blob/go1.18.10/test/run.go#L2204-L2264. Our run.go is actually based on the old version of that upstream script and could use some updates. But specifically for this function, I don't feel too motivated to write our own tests for it...